presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -656,6 +863,31 @@ private List<Subfield> findSubfields(String rootName)
                         .filter(subfield -> rootName.equals(subfield.getRootName()))
                         .collect(toImmutableList());
             }
+
+            public void setLambdaSubfields(Optional<Set<Subfield>> lambdaSubfields)
+            {
+                this.lambdaSubfields = lambdaSubfields;
+            }
+
+            public Optional<Set<Subfield>> getLambdaSubfields()
+            {
+                return lambdaSubfields;
+            }
+
+            private void giveUpOnCollectingLambdaSubfields()
```

Comment:
my understanding is that ultimately (after we're done collecting subfields), Optional.empty() and an empty set are the same (we don't add any subfields to prune).  The difference is that while we're still collecting, we don't want to see an empty set and then some other place provides a specific subfields, and so we think we actually just need that subfield instead of all subfields. 
Could we instead add the full prefix of the thing that we need all the subfields from (e.g. my_array.[*]) and then we wouldn't need this Optional representation.

Reply:
Removed Optional representation of lambda subfields.